src/core/engines/engine_manager.cpp

```diff
@@ -0,0 +1,179 @@
+#include "engine_manager.hpp"
+
+#include "core/ui/mainui.hpp"
+#include "engines/drums/drum-sampler/drum-sampler.hpp"
+#include "engines/drums/simple-drums/simple-drums.hpp"
+#include "engines/synths/nuke/nuke.hpp"
+#include "services/state.hpp"
+
+namespace otto::engines {
+  namespace {
+    std::map<std::string, std::function<AnyEngine*()>> engineGetters;
+    audio::ProcessBuffer<1> _audiobuf1;
+
+    EngineDispatcher<engines::EngineType::synth> synth;
+    EngineDispatcher<engines::EngineType::drums> drums;
+    EngineDispatcher<engines::EngineType::effect> effect;
+    Tapedeck tapedeck;
+    Mixer mixer;
+    Metronome metronome;
+    InputSelector selector;
+  } // namespace
+
+  void init()
+  {
+    engineGetters = {{"TapeDeck", [&]() { return (AnyEngine*) &tapedeck; }},
+                     {"Mixer", [&]() { return (AnyEngine*) &mixer; }},
+                     {"Synth", [&]() { return (AnyEngine*) &*synth; }},
+                     {"Drums", [&]() { return (AnyEngine*) &*drums; }},
+                     {"Metronome", [&]() { return (AnyEngine*) &metronome; }}};
+
+    register_engine<DrumSampler>();
```

Comment:
this is not really optimal i think, but it works for now, so lets roll with it.
Firstly, the map should probably own the actual engines.
It would also be cool if this engine table could maybe be defined elsewhere, but thats a bit further off even.

Reply:
sorry, forgot we moved the audiomanager etc here. The map obviously shouldnt own the engines.